flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/changelog/state/ChangelogAggregatingState.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.changelog.state;
```

Comment:
`package org.apache.flink.state.changelog;`
?

Reply:
I do not mind, either way. Let's maybe decide this after your API changes are finalized as well?

Reply:
They are now merged.

Reply:
I put state after changelog because I think these are state-related part of changelog

There may be other parts of changelog that may not be state related, for example writer/client, e.t.c

What is your reason to suggest to move the way around?

Reply:
I think that changelog is only used by backend. Theoretically, backend can use some other mechanism and changelog can be used by other parts.

For example, changelog is not mandatory to wrap incremental heap backend (only the materialization part is required).

Reply:
Where ChangelogWriter/client put (currently, in-memory is put in runtime), but what Fs-based is put?
Are they going to be put in different module than changelog state backend?

Reply:
Yes, it will be a different module.